### ISSUE TITLE
"Validate related assets" only executed after 2nd Save

### ISSUE THREAD

**Issue by tsawitzki** [craftcms/cms#7112]
### Description

In my CP I have a few tabs within a section’s entry fields. In a few of those tabs I only have one asset field. Those asset fields all have the "Validate related assets" option turned on. The individual asset volumes these asset fields refer to have a bunch of fields associated with them (categories, text, etc.) that are partly set to required. If I batch-upload a few assets to one of the fields I can save the entry _one time_ after the image thumbnails appear within the field. Only after trying to save the entry a second time or re-entering it will the assets be marked red and the expected "Couldn't save entry" message appears.

### Steps to reproduce

1. Create asset volume with a few required fields
2. Create asset field that refers to the volume created and check "Validate related assets" ("Allow uploading directly to the field" is unchecked)
3. Create entry type and associate field with it in an individual Tab
4. Go to entry with this entry type and go to the Tab
5. Choose "Add …" from asset field, throw a few asset files from HD into CP asset window and mark all
6. Click lower right "Select" button to confirm selection
7. Image thumbnails appear in the entry field, black type underneath
8. Click "Save" on entry
9. Witness entry being saved without error message
10. Open entry again and go to Tab
11. Witness all images having red type underneath
12. Hitting "Save" will throw expected message "Couldn't save entry"

### Additional info

- Craft version: Craft Solo 3.5.15.1
- PHP version: 7.2
- Database driver & version: MySQL 5.6.38
- Plugins & versions: None

**Comment by brandonkelly** [comment]
I’m not able to reproduce this.

What are each of the settings of your Assets field?

**Comment by tsawitzki** [comment]
First of all, in the Field Layout for the Asset, all fields are marked as mandatory (3 in total).

Then in the field it has checked/selected:
- "Restrict assets to a single folder" checked
- "Restrict allowed file types" checked (different checks according to field)
- View Mode "Large thumbnails"
- "Custom Selection Label" text entered
- "Validate related assets" checked

Just experienced the bug again after creating another Tab today catering to PDF assets.

**Comment by brandonkelly** [comment]
Does your Asset Location setting specify a subfolder path? If so, what?

**Comment by tsawitzki** [comment]
**Base URL** is set to `@web/assets/archive/` and **File System Path** is set to `assets/archive/`

**Comment by brandonkelly** [comment]
I was referring to the Asset Location setting in your Assets field’s settings.

**Comment by tsawitzki** [comment]
It's set to my only available folder once " Restrict assets to a single folder" is checked, then it's set to go to a sub-folder according to the content e.g. `{slug}/photography`

**Comment by brandonkelly** [comment]
Huh, I’m not able to reproduce this. I’m using the same settings locally, and have followed your steps to reproduce, but I’m never able to save the entry if any of the required asset fields aren’t filled in.

I’m not sure what else to try. Can you please send your database backup and Composer files into support@craftcms.com, and we can try to look into it on our end?

**Comment by tsawitzki** [comment]
Hey @brandonkelly, Oli wrote me he could replicate it and that you'd be further investigating on this, posting updates here, is that correct?

**Comment by brandonkelly** [comment]
@tsawitzki Yeah, I’ve also been able to reproduce now. We’re still looking into a fix though.
